### PR TITLE
Create Summary view (AI-2210)

### PR DESCRIPTION
**Pull request by annloouise**

### CODE REVIEW

File: src/enigma/models.js [hunk 1]
```diff
@@ -27,6 +27,243 @@ export async function getField(fieldName, stateName = '$') {
...
+          qSortbyYValue: 0,
+          qColumnOrder: [],
+        },
+      },
+    },
+  };
+  await init();
+  return doc.getOrCreateObject(pivotTableDef);
+}
+
```
Comment (snurrfint) [inline src/enigma/models.js]: Maybe we should try to remove some of these options. It's quite a big def. I assume many of the properties are defualt values.

Comment (annloouise) [reply src/enigma/models.js]: Done!

File: src/views/api-changes/api-changes.js [hunk 2]
```diff
@@ -179,18 +199,122 @@ customElements.define('view-api-changes', class extends AIHTMLElement {
...
+      case 'Overview':
+        return this.getOverviewContent();
+      case 'Details of updates':
+        return this.getDetailsContent();
+      default:
+        return '';
+    }
+  }
+
+  async getSummaryContent() {
```
Comment (snurrfint) [inline src/views/api-changes/api-changes.js]: Maybe we should create view components for each tab. This file is pretty big as it is already.

We can do it in the next story if you want.

File: src/views/api-changes/api-changes.js [hunk 3]
```diff
@@ -179,18 +199,122 @@ customElements.define('view-api-changes', class extends AIHTMLElement {
...
+            tempArray.shift();
+          }
+          rowData.push(colData);
+        }
+        rowDataByColumns.push(rowData);
+      });
+
+      return html`
+        <div class="summary-wrapper">
+          <div class="legend-wrapper">
```
Comment (snurrfint) [inline src/views/api-changes/api-changes.js]: Legend could also be a super small component, I think we might need it in details view as well?

File: src/views/api-changes/api-changes.js [hunk 4]
```diff
@@ -179,18 +199,122 @@ customElements.define('view-api-changes', class extends AIHTMLElement {
...
+
+    const data = layout.qHyperCube.qPivotDataPages[0]
+      ? layout.qHyperCube.qPivotDataPages[0] : null;
+
+    if (data) {
+      const tblHeaders = new Set(data.qTop.map(el => (el.qText || '')));
+      const rowHeaders = new Set(data.qLeft.map(el => (el.qText || '')));
+      const rowDataByColumns = [];
+
+      // Dividing the data by columns
```
Comment (snurrfint) [inline src/views/api-changes/api-changes.js]: Could be it's own function devideToColumns(data). Makes it testable too.

**Pull request merged**